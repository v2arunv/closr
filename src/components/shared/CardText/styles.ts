import {StyleSheet} from "react-native";
import common from "../../../common/styles";

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 10,
        backgroundColor: '#FFFFFF',
        borderBottomColor: common.borderColor,
        borderBottomWidth: 1,
    }

});

export default styles;