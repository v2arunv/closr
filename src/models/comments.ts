
export interface IComment {
    postId: number,
    id: number,
    name: string,
    email: string,
    body: string,
}

export const convertPayloadToModel = (payload: any) => {
    const {
        postId,
        id,
        name,
        email,
        body,
    } = payload;
    return <IComment> {
        postId,
        id,
        name,
        email,
        body
    };
};
