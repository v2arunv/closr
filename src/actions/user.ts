import {IUser} from "../models/users";

export interface IUserAction {
    type: string,
    user: IUser,
}

export const setUser = (user: IUser): IUserAction => {
    return <IUserAction> {
        type: 'SET_USER',
        user: user,
    }
};