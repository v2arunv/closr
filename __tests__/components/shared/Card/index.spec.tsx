import React from 'react';
import Card from '@sharedComponents/Card';
import renderer from 'react-test-renderer';
import user from '@mocks/users';
import comment from '@mocks/comments'
import CardText from "@sharedComponents/CardText";
import {IComment} from "@models/comments";

const comments: Array<IComment> = [ comment ];
test('renders correctly', () => {
    const tree = renderer.create(
        <Card user={user}
              comments={comments}
              onProfileClick={() => {}}
        >
            <CardText>
            </CardText>
        </Card>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});