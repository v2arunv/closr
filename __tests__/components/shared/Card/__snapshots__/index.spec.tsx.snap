// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#dfdfdf",
        "borderBottomWidth": 0,
        "flex": 1,
        "marginTop": 10,
        "minHeight": 150,
      },
      Object {
        "borderBottomColor": "#ddd",
        "borderBottomWidth": 0,
        "elevation": 1,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#dfdfdf",
          "borderBottomWidth": 0.5,
          "flex": 1,
          "flexDirection": "row",
          "minWidth": 100,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingTop": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderRadius": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          defaultSource={
            Object {
              "testUri": "../../../src/assets/images/dp_placeholder.jpg",
            }
          }
          source={
            Object {
              "uri": "https://semantic-ui.com/images/avatar2/large/lindsay.png",
            }
          }
          style={
            Object {
              "borderRadius": 25,
              "height": 50,
              "width": 50,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Leanne Graham
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 5,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#dfdfdf",
          "borderBottomWidth": 1,
          "flex": 1,
          "padding": 10,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      >
        <Image
          defaultSource={
            Object {
              "testUri": "../../../src/assets/images/dp_placeholder.jpg",
            }
          }
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "borderRadius": 12.5,
              "height": 25,
              "padding": 5,
              "width": 25,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#c1d0ff",
            "borderRadius": 10,
            "marginLeft": 5,
            "padding": 5,
            "width": "70%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          id labore ex et quam laborum
        </Text>
        <Text
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium
        </Text>
      </View>
    </View>
  </View>
</View>
`;
